browser/ai_chat/tools/target_util.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/tools/target_util.h"
+
+#include "base/notreached.h"
+#include "brave/components/ai_chat/core/browser/tools/tool_input_properties.h"
+#include "chrome/common/actor/actor_constants.h"
+
+namespace ai_chat::target_util {
+
+base::Value::Dict TargetProperty(const std::string& description) {
+  base::Value::Dict target_property;
+  target_property.Set("description", description);
+  base::Value::List* any_of = target_property.EnsureList("anyOf");
+
+  any_of->Append(ObjectProperty(
+      "DOM element identifiers of target (preferred)",
+      {{"document_identifier",
+        StringProperty("Document identifier for the target frame")},
+       {"content_node_id", IntegerProperty("DOM node ID of the target element "
+                                           "within the frame (optional)")}}));
+
+  any_of->Append(
+      ObjectProperty("Screen coordinates of target (less stable)",
+                     {{"x", NumberProperty("X coordinate in pixels")},
+                      {"y", NumberProperty("Y coordinate in pixels")}}));
+
+  return target_property;
+}
+
+base::expected<optimization_guide::proto::ActionTarget, std::string>
+ParseTargetInput(const base::Value::Dict& target_dict) {
+  // Check what targeting approaches are present
+  auto x_value = target_dict.FindDouble("x");
+  auto y_value = target_dict.FindDouble("y");
+  auto content_node_id = target_dict.FindInt("content_node_id");
+  auto* document_identifier = target_dict.FindString("document_identifier");
+
+  // Ensure exactly one approach is used
+  if (x_value && y_value && document_identifier) {
```

Comment:
should we also check `content_node_id` here?
```suggestion
  if (x_value && y_value && (document_identifier || content_node_id)) {
```